presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -164,6 +164,64 @@ public void testSTIsEmpty()
         assertFunction("ST_IsEmpty(ST_GeometryFromText('POLYGON EMPTY'))", BOOLEAN, true);
     }
 
+    @Test
+    public void testSTIsValid()
+    {
+        // empty geometries are valid
+        assertFunction("ST_IsValid(ST_GeometryFromText('POINT EMPTY'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('MULTIPOINT EMPTY'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('LINESTRING EMPTY'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('MULTILINESTRING EMPTY'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('POLYGON EMPTY'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('MULTIPOLYGON EMPTY'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('GEOMETRYCOLLECTION EMPTY'))", BOOLEAN, true);
+
+        // valid geometries
+        assertFunction("ST_IsValid(ST_GeometryFromText('POINT (1 2)'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('MULTIPOINT (1 2, 3 4)'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('LINESTRING (0 0, 1 2, 3 4)'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0))'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((2 4, 2 6, 6 6, 6 4)))'))", BOOLEAN, true);
+        assertFunction("ST_IsValid(ST_GeometryFromText('GEOMETRYCOLLECTION (POINT (1 2), LINESTRING (0 0, 1 2, 3 4), POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0)))'))", BOOLEAN, true);
+
+        // invalid geometries
+        assertFunction("ST_IsValid(ST_GeometryFromText('POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0), (2 2, 2 3, 3 3, 3 2, 2 2))'))", BOOLEAN, false);
+        assertFunction("ST_IsValid(ST_GeometryFromText('POLYGON ((0 0, 0 1, 2 1, 1 1, 1 0, 0 0))'))", BOOLEAN, false);
+        assertFunction("ST_IsValid(ST_GeometryFromText('POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0), (0 1, 1 1, 0.5 0.5, 0 1))'))", BOOLEAN, false);
+        assertFunction("ST_IsValid(ST_GeometryFromText('MULTIPOLYGON (((0 0, 0 1, 1 1, 1 0, 0 0)), ((0.5 0.5, 0.5 2, 2 2, 2 0.5, 0.5 0.5)))'))", BOOLEAN, false);
+        assertFunction("ST_IsValid(ST_GeometryFromText('GEOMETRYCOLLECTION (POINT (1 2), POLYGON ((0 0, 0 1, 2 1, 1 1, 1 0, 0 0)))'))", BOOLEAN, false);
+    }
+
+    @Test
+    public void testInvalidReason()
```

Comment:
We can shorten these two tests. For the valid geometries you can add a helper method (`assertValidGeometry`) to assert that they are valid AND their invalid reasons are null, so you can assert that the reason is null in `testSTIsValid` above in one go. And in `testInvalidReason` you can just test invalid geometries.